docs/articles/user_acquisition_analytics.md

```diff
@@ -0,0 +1,448 @@
+
+```python
+from huggingface_hub import notebook_login
+notebook_login() 
+```
+
+Now, let's get going on our installations and imports.
+
+First, we install superlinked and umap.
+
+```python
+%pip install superlinked==9.32.1
+%pip install umap-learn
+```
+
+Next, import all our components and constants.
+
+(Note: Omit `alt.renderers.enable(“mimetype”)` if you’re running this in [google colab](https://colab.research.google.com/github/superlinked/superlinked/blob/main/notebook/recommendations_e_commerce.ipynb). Keep it if you’re executing in [github](https://github.com/superlinked/VectorHub/blob/main/docs/articles/ecomm-recys.md).)
+
+```python
+from datetime import datetime, timedelta
+import os
+import sys
+import altair as alt
+import numpy as np
+import pandas as pd
+from sklearn.cluster import HDBSCAN
+import umap
+
+from superlinked.evaluation.charts.recency_plotter import RecencyPlotter
+from superlinked.evaluation.vector_sampler import VectorSampler
+from superlinked.framework.common.dag.context import CONTEXT_COMMON, CONTEXT_COMMON_NOW
+from superlinked.framework.common.dag.period_time import PeriodTime
+from superlinked.framework.common.embedding.number_embedding import Mode
+from superlinked.framework.common.schema.schema import Schema
+from superlinked.framework.common.schema.schema_object import String, Float, Timestamp
+from superlinked.framework.common.schema.id_schema_object import IdField
+from superlinked.framework.common.parser.dataframe_parser import DataFrameParser
+from superlinked.framework.dsl.executor.in_memory.in_memory_executor import (
+    InMemoryExecutor,
+    InMemoryApp,
+)
+from superlinked.framework.dsl.index.index import Index
+from superlinked.framework.dsl.source.in_memory_source import InMemorySource
+from superlinked.framework.dsl.space.text_similarity_space import TextSimilaritySpace
+from superlinked.framework.dsl.space.number_space import NumberSpace
+from superlinked.framework.dsl.space.recency_space import RecencySpace
+
+alt.renderers.enable("mimetype")  # important: to render plots in colab, replace 'mimetype' with 'colab'
+alt.data_transformers.disable_max_rows()
+os.environ["TOKENIZERS_PARALLELISM"] = "false"
+pd.set_option("display.max_colwidth", 190)
+pd.options.display.float_format = "{:.2f}".format
+```
+
+Now we import our dataset. 
+
+```python
+dataset_repository_url = (
+    "https://storage.googleapis.com/superlinked-notebook-user-acquisiton-analytics"
+)
+USER_DATASET_URL = f"{dataset_repository_url}/user_acquisiton_data.csv"
+NOW_TS = 1708529056
+EXECUTOR_DATA = {CONTEXT_COMMON: {CONTEXT_COMMON_NOW: NOW_TS}}
+```
+
+(This might take a few minutes. While you're waiting, you can always find other interesting reading in [VectorHub](https://superlinked.com/vectorhub/).)
```

Comment:
the previous code just sets some defaults and does not do any loading (therefore it is fast)
source.put takes some time, and that is the place where I believe I've put this message